src/lib/helpers.ts

```diff
@@ -119,7 +119,10 @@ export const convertGravityToConnectionArgs = <
 
 export const removeNulls = (object) => {
   Object.keys(object).forEach(
-    (key) => object[key] == null && delete object[key]
+    (key) =>
+      (object[key] == null ||
+        (Array.isArray(object[key]) && isEmpty(object[key]))) &&
```

Comment:
probably the wrong place for this, but we want to remove empty arrays here(!). This allows us to cleanly merge filter params together, without an empty array (coming from the filter), overwriting a filled-in array (coming from the KAWS `query` during stitching)